api/src/main/java/org/apache/iceberg/io/CredentialSupplier.java

```diff
@@ -26,7 +26,7 @@
  * the configured credential as a string, and use the credential for file access via other IO
  * libraries.
  */
-public interface CredentialSupplier {
+public interface CredentialSupplier extends FileIO {
```

Comment:
All these interfaces assume they are mixed-in by `FileIO` instances (see docs).

Reply:
This is not quite correct. `CredentialSupplier` may extend other interfaces and not just `FileIO`.

Reply:
You are right. I misinterpreted the doc.